src/core/src/sim/mc/monte_carlo.rs

```diff
@@ -64,10 +67,21 @@ impl MonteCarlo {
                 we can not add new moves."
             );
         }
-        self.moves.push(mcmove);
+        self.moves.push((mcmove, MoveCounter::new(None)));
         self.frequencies.push(frequency);
     }
 
+    pub fn add_with_acceptance(&mut self, mcmove: Box<MCMove>, frequency: f64, target: Option<f64>) {
```

Comment:
The `target` parameter should be a f64 here. there is already a function for passing `None`.